basis/english/english-docs.factor

```diff
@@ -0,0 +1,209 @@
+{ $description "Gives the proper indefinite singular article (" { $emphasis "a" } " or " { $emphasis "an" } ") for the word. For words which begin with a vowel sound, " { $emphasis "an" } " is used, whereas " { $emphasis "a" } " is used for words which begin with a consonant sound." }
+{ $notes "The output does not contain the input. The output of this word is always a singular article, regardless of the plurality of the input." }
+{ $examples
+    { $example
+        "USING: english kernel combinators sequences io ;"
+        "\"object\" [ a/an ] keep \" \" glue print"
+        "an object"
+    }
+} ;
+
+HELP: ?plural-article
+{ $values { "word" string } { "article" { $or-markup-example "\"a\"" "\"an\"" "\"the\"" } } }
+{ $description "Output the proper article given the plurality and first letter of the input. Unlike " { $link a/an } " this word handles plural inputs by outputting the definite " { $emphasis "\"the\"" } ". If the input is singular as determined by " { $link singular? } " this word operates like " { $link a/an } "." }
+{ $notes { $list "English lacks a plural indefinite article, so the plural definite is used here instead." $keep-case $0-plurality } }
+{ $examples
+    { $example
+        "USING: english sequences kernel io ;"
+        "\"cat\" [ ?plural-article ] keep \" \" glue print"
+        "a cat"
+    }
+    { $example
+        "USING: english sequences kernel io ;"
+        "\"cats\" [ ?plural-article ] keep \" \" glue print"
+        "the cats"
+    }
+} ;
+
+HELP: comma-list-by
+{ $values
+    { "parts" sequence }
+    { "quot" { $quotation ( part index -- newpart ) } }
+    { "conjunction" string }
+    { "clause-seq" sequence }
+}
+{ $description "Generate a comma-separated list of things from the result of applying " { $snippet "quot" } " to each element in "  { $snippet "parts" } ", emplacing " { $snippet "conjunction" } " before the last " { $snippet "part" } " if there are two or more elements in " { $snippet "parts" } "." }
+{ $notes $keep-case }
+{ $examples { "See the examples in " { $link $or-markup-example } " and " { $link simple-comma-list } "." } } ;
+
+HELP: simple-comma-list
+{ $values { "parts" sequence } { "conjunction" string } { "parts'" sequence } }
+{ $description "Punctuate and conjunctionize the input words into a comma-separated list construction." }
+{ $examples
+    { $example
+        "USING: english io sequences splitting ;"
+        "{ \"a cat\" \"a peach\" \"an object\" } \"or\" simple-comma-list [ \"\" join ] map \"\" join print"
+        "a cat, a peach, or an object"
+    }
+} ;
+
+HELP: or-markup-example
+{ $values { "markup" "a sequence of markup elements" } { "classes" "a sequence of words" } }
+{ $description "Used to implement " { $link $or-markup-example } " and demonstrate " { $link comma-list-by } "." }
+{ $examples { "See the examples in " { $link $or-markup-example } "." } } ;
+
+HELP: $or-markup-example
```

Comment:
Help markup maybe should be in the ``help`` vocab hierarchy rather than ``english``?

Reply:
Yes, but these words `$or-markup-example` and `or-markup-example` are only to demonstrate the use of `comma-list-by`. 

When I write the pull request which makes `help.markup` use `english`'s `a/an` then I will look into using an implementation like this for `$or` / `($instance)` so that we don't have so much natural language code in `help.markup`.